### ISSUE TITLE
args.result  in afterInsert(args) does not match results returned from query

### ISSUE THREAD

**Issue by meelash** [Vincit/objection.js#2342]
**Summary**
1. According to the objection.js documentation: https://vincit.github.io/objection.js/api/types/#type-statichookarguments, I expect the result property in afterInsert, StaticHookArguments to be the returned result from the query. When my sqlite3 schema has a column of type boolean, I get (as expected per this issue: https://github.com/Vincit/objection.js/issues/174) a value 0 or 1 for that column after fetching. I would therefore expect the same to be true in the result property of afterInsert's arguments, but instead the data in the result property has type boolean, true or false. That suggests that it is not really the result of the query being returned, but the model instance created before running the query.
2. Secondly, according to the documentation here: https://vincit.github.io/objection.js/api/model/static-methods.html#static-afterinsert I expect that if I return the result in the afterInsert function, I should get the same object returned from the insertAndFetch query. That, however, is only the case for newly added properties; modifications to existing properties are overwritten by the actual data from the query.

In summary, it seems like the data from the query is being set on the object in the result property of the StaticHookArguments after afterInsert is called instead of before.

**Reproduction**
Minimal reproduction using template is available here:
https://github.com/meelash/objection_afterInsert_bug

**Expected Behavior**

1. The result which is logged to console in the afterInsert method should have isWoman = 1, not isWoman = true;
2. testResult1 should, in addition to having property foo set, should have isWoman be false.

**Comment by meelash** [comment]
Actually, my conclusion about afterInsert being called with data from before the insert rather than the returned data is at least partially wrong, because you can see that the testResult3 that is logged to the console in the reproduction has the id property set which could only have come from the fetched data. But then why is the boolean of type boolean and not the integer that comes back? Very strange behavior.
